Update copy-delivery.md (#2388)
cal-itp/benefits: docs/product-and-design/copy-delivery.md

- In-line link URLs
- Error messages (like [no script](https://github.com/cal-itp/benefits/blob/main/benefits/core/templates/core/includes/noscript.html), [no cookies](https://github.com/cal-itp/benefits/blob/main/benefits/core/templates/core/includes/nocookies.html) warnings)

> Developers create the 'django.po' message files from copy created and implemented by Design and Product. Design and Product use the tools Figma, Ditto, and Crowdin to craft product copy, manage all copy and variantions of copy (e.g. agency-specific copy or Spanish translation copy), and translate copy.

Reviewer: ```suggestion
Developers update `django.po` message files for each supported langauge from copy created and implemented by Design and Product. Design and Product use Figma, Ditto, and Crowdin to craft product copy, manage all copy and variantions of copy (e.g. agency-specific copy or Spanish translation copy), translate copy, and ultimately deliver production-ready copy to Developers via Figma.
```

Ditto is our copy management system for Benefits and the Cal-ITP sites. It syncs directly with Figma and uses many of the same conventions found in Figma, including components and variables (therefore, it is important to be specific when referring to a Ditto component or a Figma variable).

The Benefits project within Ditto is fully synced with Figma. Product and design may use either Figma or Ditto as their source of truth on copy, though it is recommended that Developers use Figma for all final copy. Compiler is considering using the Github integration that comes with Ditto, but has not formally integrated Ditto into their workflow as of yet.

In the Benefits project, all of the text has been componentized. Agencies are represented in the text as variants on components. Languages are also represented in the text as variants. Currently Benefits has two languages: English and Spanish. English is the assumed default in the copy in Ditto and Spanish is a variant. There are variants within Ditto that include both an agency and Spanish, such as MST-Spanish.

### Crowdin

> Crowdin is our translation platform. We use iBabbleon to provide our translations. With very simple translations (or more complex if you are fluent or native in the language), Compiler team members can translate the strings of text directly in the Crowdin platform. For anything above our language skills we contact Salim at iBabbleon and ask for the new files to be translated once the language files have been synced from Ditto to Crowdin.

Reviewer: ```suggestion
Crowdin is our translation platform. We use iBabbleon to provide our translations. With very simple translations (or more complex if you are fluent or native in the language), Compiler team members can translate the strings of text directly in the Crowdin platform. For anything above our language skills, Design contacts Salim at iBabbleon and ask for the new files to be translated once the language files have been synced from Ditto to Crowdin.
```

## Copy delivery process responsibilities

### Product

- Engage with product stakeholders to get the information necessary for copy writing.
- Collaborate with Design on copy writing.
- Help verify English and Spanish copy is ready for Engineering.

### Design

- Draft, proof and ready the copy for review.
- Engage with client editorial/communications team to ensure English language and Spanish language copy are edited according to client style guides.

> - Ensure copy is synced between Figma, Ditto, and Crowdin

Reviewer: ```suggestion
- Ensure copy is synced between Figma, Ditto, and Crowdin.
```

> - Engage with the translation agency, [iBabbleOn](https://ibabbleon.com/), to get Spanish translations ready for Engineering.

Reviewer: ```suggestion
- Engage with the translation agency, [iBabbleOn](https://ibabbleon.com/), to get Spanish translations ready for Engineering.
- Deliver production-ready copy, with Spanish translations, to Engineering at Design Hand-Off meeting via a synced Figma file in the main branch.
```

- In-line link URLs
- Error messages (like [no script](https://github.com/cal-itp/benefits/blob/main/benefits/core/templates/core/includes/noscript.html), [no cookies](https://github.com/cal-itp/benefits/blob/main/benefits/core/templates/core/includes/nocookies.html) warnings)

Developers create the 'django.po' message files from copy created and implemented by Design and Product. Design and Product use the tools Figma, Ditto, and Crowdin to craft product copy, manage all copy and variantions of copy (e.g. agency-specific copy or Spanish translation copy), and translate copy.

## Tools used in Cal-ITP Benefits Application Copy

### Figma

Figma is our primary design tool and is the source of truth for all design decisions, concepts, and directions for the Benefits app and all Cal-ITP sites. We have Figma Organization which includes developer mode, version history, and branching–a tool to explore new design directions.

> The development team prefers not to have admin or editor capabilities so as not to accidentally interfere with any designs. Currently there is no formal process within Figma for handoff. Instead handoffs occur over the course of two meetings, a review of the designs and then what we call “issue-fest,” or a chance for developers to create Github issues while designers can weigh in and answer questions.

Reviewer: ```suggestion
The development team prefers not to have admin or editor capabilities so as not to accidentally interfere with any designs. Currently there is no formal process within Figma for handoff. Instead handoffs occur over the course of two meetings, a review of the designs and then what we call “issue-fest,” or a chance for developers to create GitHub issues while designers can weigh in and answer questions.
```

@srhhnry Aren't we gonna use the Ready for dev feature in Figma for handoff? 
<img width="280" alt="image" src="https://github.com/user-attachments/assets/1ef5437e-7152-461b-ae73-373072eda594">

- If anyone (Product, Engineering) has concerns about copy, should they put a comment on Ditto, on Figma, or some other way?

Reviewer: If your preference is for comments in Figma, then we can make that the policy. You can note that all comments on copy or design should be made in Figma, and that all comments need to be submitted by the Hand-Off meeting. No comments should be added after the Hand-Off meeting ( as in, engineers cannot ask design to change anything _after_ hand-off. that would be a new design process ).

The dev team and design largely collaborate on designs through Slack and comments directly in Figma. Design aims to address Figma comments as soon as possible or within one day.

### Ditto

The human-readable version of the English and Spanish translation strings for the application are delivered to Product and Engineering by Design, and are managed in the Ditto application. [Cal-ITP Benefits](https://app.dittowords.com/projects/65e119db3f806ee92198378a/page/5766:22626).

Ditto is our copy management system for Benefits and the Cal-ITP sites. It syncs directly with Figma and uses many of the same conventions found in Figma, including components and variables (therefore, it is important to be specific when referring to a Ditto component or a Figma variable).

> The Benefits project within Ditto is fully synced with Figma. Product and design may use either Figma or Ditto as their source of truth on copy, though it is recommended that Developers use Figma for all final copy. Compiler is considering using the Github integration that comes with Ditto, but has not formally integrated Ditto into their workflow as of yet.

Reviewer: Question for @srhhnry - It makes sense to me for developers to use Figma to copy and paste English copy, but for getting Spanish copy, we'd have to go into Ditto, right? Or is there a way to check Figma for Spanish copy?

Author: - For viewing copy in Spanish in Figma, one needs to have the Ditto plugin active. From there, select a frame and the Ditto plugin should automatically move to the Selected tab, and within that tab there's a dropdown to preview different variants. 

<img width="1271" alt="image" src="https://github.com/user-attachments/assets/616e52aa-a582-4fda-98c7-9c7d4ee1d77c">

Everyone has varying levels of permission, so if it is not possible to view that tab within the Ditto plugin, then yes I recommend using Ditto. 

(should the above^  go into the documentation as well?)

Reviewer: This is _exactly_ the kind of instruction that should go in the Documentation. A good rule of thumb is: If you think you might have to copy/paste the same instructions to more than 1 person, it should be in the Docs. You could format is as 1, 2, 3 numbered instructions.

> The Benefits project within Ditto is fully synced with Figma. Product and design may use either Figma or Ditto as their source of truth on copy, though it is recommended that Developers use Figma for all final copy. Compiler is considering using the Github integration that comes with Ditto, but has not formally integrated Ditto into their workflow as of yet.

Reviewer: ```suggestion
The Benefits project within Ditto is fully synced with Figma. Product and design may use either Figma or Ditto as their source of truth on copy, though it is recommended that Developers use Figma for all final copy. Compiler is considering using the GitHub integration that comes with Ditto, but has not formally integrated Ditto into their workflow as of yet.
```
